src/platform/silabs/wifi/rs911x/WifiInterface.cpp

```diff
@@ -677,68 +677,60 @@ void ProcessEvent(WifiPlatformEvent event)
 
-        if (wfx_rsi.scan_cb == NULL)
+        uint8_t nbreOfScannedNetworks = scan_rsp.scan_count[0];
+        for (int i = 0; i < nbreOfScannedNetworks; i++)
         {
-            ChipLogError(DeviceLayer, "wfx_rsi.scan_cb is NULL");
-            return;
-        }
+            rsi_scan_info_t scan      = scan_rsp.scan_info[i];
+            wfx_wifi_scan_result_t ap = { 0 };
 
-        rsi_scan_info_t * scan;
-        wfx_wifi_scan_result_t ap;
+            ap.ssid_length = strnlen(reinterpret_cast<char *>(scan.ssid), WFX_MAX_SSID_LENGTH);
 
-        for (int x = 0; x < scan_rsp.scan_count[0]; x++)
-        {
-            scan = &scan_rsp.scan_info[x];
-            // clear structure and calculate size of SSID
-            memset(&ap, 0, sizeof(ap));
-            ap.ssid_length =
-                strnlen(reinterpret_cast<char *>(scan->ssid), std::min<size_t>(sizeof(scan->ssid), WFX_MAX_SSID_LENGTH));
-            chip::Platform::CopyString(ap.ssid, ap.ssid_length, reinterpret_cast<char *>(scan->ssid));
-
-            // check if the scanned ssid is the one we are looking for
-            if (wfx_rsi.scan_ssid_length != 0 && strncmp(wfx_rsi.scan_ssid, ap.ssid, WFX_MAX_SSID_LENGTH) != 0)
+            chip::ByteSpan scannedSsid(scan.ssid, ap.ssid_length);
+            chip::MutableByteSpan outputSsid(ap.ssid, WFX_MAX_SSID_LENGTH);
+            chip::CopySpanToMutableSpan(scannedSsid, outputSsid);
+
+            // Check if the scanned ssid is the requested Ssid
+            chip::ByteSpan requestedSsid(wfx_rsi.scan_ssid, wfx_rsi.scan_ssid_length);
+            if (!requestedSsid.empty() && !requestedSsid.data_equal(scannedSsid))
             {
                 continue; // we found the targeted ssid.
             }
+
             // TODO: convert security mode from RSI to WFX
-            ap.security = static_cast<wfx_sec_t>(scan->security_mode);
-            ap.rssi     = (-1) * scan->rssi_val;
+            ap.security = static_cast<wfx_sec_t>(scan.security_mode);
+            ap.rssi     = (-1) * scan.rssi_val;
 
             configASSERT(sizeof(ap.bssid) == kWifiMacAddressLength);
-            configASSERT(sizeof(scan->bssid) == kWifiMacAddressLength);
-            memcpy(ap.bssid, scan->bssid, kWifiMacAddressLength);
-            (*wfx_rsi.scan_cb)(&ap);
+            configASSERT(sizeof(scan.bssid) == kWifiMacAddressLength);
+
+            chip::MutableByteSpan bssidSpan(ap.bssid, kWifiMacAddressLength);
+            chip::ByteSpan scanBssidSpan(scan.bssid, kWifiMacAddressLength);
+            chip::CopySpanToMutableSpan(scanBssidSpan, bssidSpan);
+
+            wfx_rsi.scan_cb(&ap);
 
-            // no ssid filter set, return all results
-            if (wfx_rsi.scan_ssid_length == 0)
+            // If we reach this and the requestedSsid is not empty, it means we found the requested SSID and we can exit
+            if (!requestedSsid.empty())
             {
-                continue;
+                break;
```

Comment:
Is this where we should have the `// we found the targeted ssid.` comment?

Reply:
Previous comment was incorrect.